modules/bootstrap-launcher/src/it/scala/coursier/bootstrap/launcher/DownloadTests.scala

```diff
@@ -0,0 +1,105 @@
+package coursier.bootstrap.launcher;
+
+import utest._
+
+import coursier.bootstrap.launcher.credentials.Credentials
+import coursier.bootstrap.launcher.credentials.DirectCredentials
+
+import scala.jdk.CollectionConverters._
+import java.nio.file.Files
+import java.io.File
+import java.nio.file.Path
+import java.net.URL
+import java.nio.file.Paths
+import java.net.URI
+
+object DownloadTests extends TestSuite {
+  
+  private val testRepository = Option(System.getenv("TEST_REPOSITORY")).getOrElse(sys.error("TEST_REPOSITORY not set"))
+  private val testRepositoryUser = Option(System.getenv("TEST_REPOSITORY_USER")).getOrElse(sys.error("TEST_REPOSITORY_USER not set"))
+  private val testRepositoryPassword = Option(System.getenv("TEST_REPOSITORY_PASSWORD")).getOrElse(sys.error("TEST_REPOSITORY_PASSWORD not set"))
+  private val testRepositoryHost = new URI(testRepository).getHost
+
+  private def deleteRecursive(f: File): Unit = {
+    if (f.isDirectory)
+      f.listFiles().foreach(deleteRecursive)
+    if (f.exists())
+      f.delete()
+  }
+
+  private def withTmpDir[T](f: Path => T): T = {
+    val dir = Files.createTempDirectory("coursier-test")
+    val shutdownHook: Thread =
+      new Thread {
+        override def run() =
+          deleteRecursive(dir.toFile)
+      }
+    Runtime.getRuntime.addShutdownHook(shutdownHook)
+    try f(dir)
+    finally {
+      deleteRecursive(dir.toFile)
+      Runtime.getRuntime.removeShutdownHook(shutdownHook)
+    }
+  }
+
+  val tests = Tests {
+
+    test("public URL") {
+      withTmpDir { dir =>
+        try {
+          Download.setCacheDir(dir.toFile)
```

Comment:
I think the tests might run in parallel, which would be a problem here… To work around that, we can remove the static initializer on `Download`, and make its methods non-static. https://github.com/coursier/coursier/pull/2000 does that, don't hesitate to rebase your work on it.

It should allow to just do
```scala
val download = new Download(dir.toFile, …)
download.getLocalURLs(…)
```
so that each test has its own `Download` instance.

Reply:
This is fine with me. I see that the `bootstrap-launcher` doesn't enforce source and binary compatibility. Are you going to merge #2000? Then I'll rebase.

Thanks

Reply:
There's no binary compatibility checks for `bootstrap-launcher`, as it's not meant to be used as a library (the same applies to the `cli` module). I just merged #2000, just feel free to rebase.

Reply:
That's done: branch rebased and review comments applied. Please let me know if I missed anything. Thanks